owncloudApp/src/androidTest/java/com/owncloud/android/database/MigrationTest.kt

```diff
@@ -0,0 +1,187 @@
+import com.owncloud.android.db.ProviderMeta.ProviderTableMeta
+import com.owncloud.android.db.ProviderMeta.ProviderTableMeta.CAPABILITIES_ACCOUNT_NAME
+import com.owncloud.android.db.ProviderMeta.ProviderTableMeta.CAPABILITIES_CORE_POLLINTERVAL
+import com.owncloud.android.db.ProviderMeta.ProviderTableMeta.CAPABILITIES_SHARING_PUBLIC_EXPIRE_DATE_DAYS
+import com.owncloud.android.db.ProviderMeta.ProviderTableMeta.CAPABILITIES_VERSION_MAYOR
+import com.owncloud.android.db.ProviderMeta.ProviderTableMeta.CAPABILITIES_VERSION_MICRO
+import com.owncloud.android.db.ProviderMeta.ProviderTableMeta.CAPABILITIES_VERSION_MINOR
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.IOException
+
+@RunWith(AndroidJUnit4::class)
+@SmallTest
+class MigrationTest {
+
+    @Rule
+    @JvmField
+    val helper: MigrationTestHelper = MigrationTestHelper(
+        InstrumentationRegistry.getInstrumentation(),
+        OwncloudDatabase::class.java.canonicalName,
+        FrameworkSQLiteOpenHelperFactory()
+    )
+
+    @Test
+    fun startInVersion27_containsCorrectData() {
+        // Create the database with version 27
+        with(
+            helper.createDatabase(
+                TEST_DB_NAME,
+                DB_VERSION_27
+            )
+        ) {
+
+            // Insert some data
+            insert(ProviderTableMeta.CAPABILITIES_TABLE_NAME, SQLiteDatabase.CONFLICT_NONE, cv)
+
+            // Close database
+            close()
+        }
+
+        // Verify that the data is correct
+        val dbCapability =
+            getMigratedRoomDatabase().capabilityDao().getCapabilityForAccount(cv.getAsString(CAPABILITIES_ACCOUNT_NAME))
+        assertEquals(dbCapability.accountName, cv.getAsString(CAPABILITIES_ACCOUNT_NAME))
+        assertEquals(dbCapability.versionMayor, cv.getAsInteger(CAPABILITIES_VERSION_MAYOR))
+        assertEquals(dbCapability.versionMinor, cv.getAsInteger(CAPABILITIES_VERSION_MINOR))
+        assertEquals(dbCapability.versionMicro, cv.getAsInteger(CAPABILITIES_VERSION_MICRO))
+        assertEquals(dbCapability.corePollInterval, cv.getAsInteger(CAPABILITIES_CORE_POLLINTERVAL))
+        assertEquals(
+            dbCapability.filesSharingPublicExpireDateDays,
+            cv.getAsInteger(CAPABILITIES_SHARING_PUBLIC_EXPIRE_DATE_DAYS)
+        )
+    }
+
+    @Test
+    @Throws(IOException::class)
+    fun migrate27To28() {
+        with(
+            helper.createDatabase(
+                TEST_DB_NAME,
+                DB_VERSION_27
+            )
+        ) {
+            // Database has schema version 27. Insert some values to test if they are migrated successfully.
+            // We cannot use DAO classes because they expect the latest schema and we may not have some fields there.
+            insert(ProviderTableMeta.CAPABILITIES_TABLE_NAME, SQLiteDatabase.CONFLICT_NONE, cv)
```

Comment:
Yes, and that is the problem. We cannot use DAO because it should be updated with last schema, and maybe we have new fields that on previous version we do not have. That will cause us problems there when we try to introduce rows that may have differed from one version to another.